### PR TITLE
cephfs-shell: chmod should be allowed for changing all mode bits.

### PR DESCRIPTION
**Pull request by neesingh-rh**
The allowable mask is changed to 07777 for chmod.

Fix: https://tracker.ceph.com/issues/48863
Signed-off-by: Neeraj Pratap Singh <neesingh@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/tools/cephfs/cephfs-shell [hunk 1]
```diff
@@ -552,11 +550,11 @@ class CephFSShell(Cmd):
...
-                parser.error("invalid mode: %s\n"
-                             "mode cannot be negative" % values)
-            if o_mode > 0o777:
-                parser.error("invalid mode: %s\n"
-                             "mode cannot be greater than octal 0777" % values)
+                parser.error(f"invalid mode: {values}\n"
+                             "mode cannot be negative")
+            if o_mode > 0o7777:
+                parser.error(f"invalid mode: {values}\n"
+                             "mode cannot be greater than octal 0777")
```
Comment (jtlayton) [inline src/tools/cephfs/cephfs-shell:557]: The error message needs to be updated here.

**Review by jtlayton — APPROVED** [review]

File: src/tools/cephfs/cephfs-shell (hunk 1 above)
Comment (neesingh-rh) [reply src/tools/cephfs/cephfs-shell]: Done!!

**Review by jtlayton — APPROVED** [review]

**Comment by neesingh-rh** [comment]
jenkins test make check

**Review by dparmar18 — APPROVED** [review]
Looks good

**Review by nmshelke — APPROVED** [review]
LGTM.

**Review by vshankar — COMMENTED** [review]
Needs tests added.

**Comment by neesingh-rh** [comment]
> Needs tests added.

Test is added.

**Comment by neesingh-rh** [comment]
jenkins test windows

**Review by vshankar — CHANGES_REQUESTED** [review]
@neesingh-rh please take a look at the failures (they look related):

https://pulpito.ceph.com/vshankar-2022-04-19_09:09:31-fs:shell-wip-vshankar-testing1-20220419-121051-testing-default-smithi/

**Comment by neesingh-rh** [comment]
> @neesingh-rh please take a look at the failures (they look related):
> 
> https://pulpito.ceph.com/vshankar-2022-04-19_09:09:31-fs:shell-wip-vshankar-testing1-20220419-121051-testing-default-smithi/

Fixed!!

**Comment by neesingh-rh** [comment]
jenkins test api

**Comment by neesingh-rh** [comment]
jenkins test make check

**Comment by vshankar** [comment]
> > @neesingh-rh please take a look at the failures (they look related):
> > https://pulpito.ceph.com/vshankar-2022-04-19_09:09:31-fs:shell-wip-vshankar-testing1-20220419-121051-testing-default-smithi/
> 
> Fixed!!

Looks good!

**Review by vshankar — APPROVED** [review]

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2022-May-04

**Pull request merged**